exercise/templates/exercise/staff/_submissions_table.html

```diff
@@ -6,16 +6,31 @@
 
 <div class="clearfix">
   <div class="pull-right">
-    <p>
-      <a class="btn btn-default btn-sm" href="{{ exercise|url:'submission-summary' }}">
-        <span class="glyphicon glyphicon-stats" aria-hidden="true"></span>
-        {% trans "Summary" %}
-      </a>
-      <a class="btn btn-default btn-sm" href="{% url 'api:course-submissiondata-list' version=2 course_id=instance.id %}?exercise_id={{ exercise.id }}&format=csv">
-        <span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span>
-        {% trans "Download CSV" %}
-      </a>
-    </p>
+    <a class="btn btn-default btn-sm" href="{{ exercise|url:'submission-summary' }}">
+      <span class="glyphicon glyphicon-stats" aria-hidden="true"></span>
+      {% trans "Summary" %}
+    </a>
+    <span class="dropdown pull-right">
+      <button
+        class="btn btn-default dropdown-toggle"
+        type="button"
+        data-toggle="dropdown"
+        id="download-data"
+      >
+        {% trans "Download" %} <span class="caret"></span>
+      </button>
+      <ul class="dropdown-menu" aria-labeledby="download-data">
+        {% with formats='csv excel.csv json' %}
+        {% for format in formats.split %}
+        <li>
+          <a href="{% url 'api:course-submissiondata-list' version=2 course_id=instance.id %}?exercise_id={{ exercise.id }}&amp;format={{ format }}">
+            <pre>{{ format }}</pre>
```

Comment:
I see this as a bit too simple solution. Maybe tuple with formats and nicer strings?

Reply:
Any ideas what would be a nice way to implement this?  I don't know how to (easily) define a list of tuples in django templates, but I don't want to do it in python because this template is used in several places and I want to avoid writing everything twice writing everything twice..

Reply:
https://github.com/django/django/blob/master/django/templatetags/i18n.py#L221

Probably template tag that injects nice list of tuples in render context, like above example from i18n tags.